pkg/appendable/appendable.go

```diff
@@ -162,8 +162,8 @@ func DetermineType(ft FieldType) uint16 {
 		width = uint16(shift + 0)
 	case FieldTypeFloat64, FieldTypeInt64, FieldTypeUint64:
 		width = uint16(shift + 8)
-	case FieldTypeTrigram:
-		width = uint16(shift + trigram.N)
+	case FieldTypeNgram:
+		width = uint16(shift + ngram.MAX_GRAM) // we'll bound all ngrams with the max len which is N = 3
```

Comment:
Is this storing different n-gram lengths in the same b-tree? I don't think we should do that because the disk space cost of storing the lengths will be very high.

If we want to store multiple length grams, we can do it by creating a separate index. The main observation is that it never makes sense to compare across n-gram lengths, like comparing a 2-gram against a 3-gram has no semantic meaning and n-grams should only be compared across the same length. Thus, making a separate index is probably more efficient.

Reply:
Yeah this makes total sense. Will make two additional indexes (1-gram and 2-gram)